### PR TITLE
[NEW] Custom user name field from Custom OAuth

### LINKED ISSUE #14380: [Feature] Custom user name field from custom OAuth
Add ability to specify which field from custom OAuth identity response should be mapped to user name field (not username which is already supported, but name).

### PR DESCRIPTION
**Pull request by mjovanovic0**
@RocketChat/core

Closes https://github.com/RocketChat/feature-requests/issues/107

In settings I added new property: Accounts_OAuth_Custom_Name_Field which holds information of field in custom OAuth identity response from which user name should be populated  (ex.: `fullName`)

### CODE REVIEW

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -94,6 +94,7 @@
   "Accounts_OAuth_Custom_Token_Path": "Token Path",
   "Accounts_OAuth_Custom_Token_Sent_Via": "Token Sent Via",
   "Accounts_OAuth_Custom_Username_Field": "Username field",
+  "Accounts_OAuth_Custom_Name_Field": "Custom Name field",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]:
```suggestion
  "Accounts_OAuth_Custom_Name_Field": "Name field",
```

**Comment by geekgonecrazy** [comment]
I've gone ahead and approved, will likely merge after we get 1.0.3 out the door.  Thanks!

**Comment by trialsin** [comment]
How about other custom field?
Like, I've a customField "lotacao". with LDAP it is possible:

image

How can I do it with custom OAUTH profile?

**Comment by rodrigok** [comment]
We can add a mapping as we do on LDAP

**Comment by trialsin** [comment]
> 
> 
> We can add a mapping as we do on LDAP

It would be nice! Some easy workaround for now?

Thank you.

**Pull request merged**